src/Microsoft.OpenApi.OData.Reader/Common/Utils.cs

```diff
@@ -222,5 +222,50 @@ internal static void AddCustomAttributesToExtensions(this IDictionary<string, IO
 
             return atrributesValueMap;
         }
+
+        /// <summary>
+        /// Checks whether the base type of an <see cref="IEdmStructuredType"/> is referenced as a type within the Edm model.
+        /// </summary>
+        /// <param name="model">The Edm model.</param>
+        /// <param name="baseType">The base type of the target <see cref="IEdmStructuredType"/>.</param>
+        /// <param name="structuredTypes">Optional: The IEnumerable of <see cref="IEdmStructuredType"/> to check against.</param>
+        /// <param name="actions">Optional: The IEnumerable of <see cref="IEdmAction"/> to check against.</param>
+        /// <returns>True if reference is found, otherwise False.</returns>
+        internal static bool IsBaseTypeReferencedAsTypeInModel(
+            this IEdmModel model,
+            IEdmStructuredType baseType,
+            IEnumerable<IEdmStructuredType> structuredTypes = null,
+            IEnumerable<IEdmAction> actions = null)
+        {
+            string baseTypeName = baseType?.FullTypeName();
+            bool isBaseTypeEntity = Constants.EntityName.Equals(baseTypeName?.Split('.').Last(), StringComparison.OrdinalIgnoreCase);
+
+            if (!string.IsNullOrEmpty(baseTypeName) && !isBaseTypeEntity)
+            {
+                structuredTypes ??= model.GetAllElements()
+                        .Where(static x => x.SchemaElementKind == EdmSchemaElementKind.TypeDefinition)
+                        .Where(static y => !y.Name.Equals(Constants.EntityName, StringComparison.OrdinalIgnoreCase))
+                        .OfType<IEdmStructuredType>();
+
+                actions ??= model.GetAllElements()
+                        .Where(static x => x.SchemaElementKind == EdmSchemaElementKind.Action)
+                        .OfType<IEdmAction>();
+
+                // Is base type referenced as a type in any property within a structured type
+                bool isReferencedInStructuredType = structuredTypes
+                    .Any(x => x.DeclaredProperties.Where(x => x.Type.TypeKind() == EdmTypeKind.Entity)
```

Comment:
```suggestion
```
I'm not sure about this check being here because of two things,
- This leaves out checking Complex types which could possibly have an abstract base yet is still a stuctured type.
- The property could be a collection property and I believe `x.Type.TypeKind()` will return `EdmTypeKind.Collection` in such a scenario.